x-pack/plugins/enterprise_search/public/applications/app_search/components/document_creation/document_creation_logic.test.ts

```diff
@@ -18,11 +32,29 @@ describe('DocumentCreationLogic', () => {
     creationStep: DocumentCreationStep.AddDocuments,
     textInput: dedent(DOCUMENTS_API_JSON_EXAMPLE),
     fileInput: null,
+    isUploading: false,
+    warnings: [],
+    errors: [],
+    summary: {},
   };
   const mockFile = new File(['mockFile'], 'mockFile.json');
 
-  const mount = () => {
-    resetContext({});
+  const mount = (defaults?: object) => {
+    if (!defaults) {
+      resetContext({});
+    } else {
+      resetContext({
+        defaults: {
+          enterprise_search: {
+            app_search: {
+              document_creation_logic: {
+                ...defaults,
+              },
+            },
+          },
+        },
+      });
+    }
```

Comment:
I'm thinking about writing a reusable test helper for this logic at some point in the future 🤔 some day!!

Reply:
+1

Reply:
I'd be made cool if it could extract the path from the logic automatically.

```
const mount = new Mounter(EngineLogic)
...
mount()
mount({ engineName: "foo" })
```

Reply:
Follow-up: https://github.com/elastic/kibana/pull/87247